codegen/builtin_string.go

```diff
@@ -53,6 +55,64 @@ func (t Template) Call(ctx context.Context, cln *client.Client, ret Register, op
 	return ret.Set(buf.String())
 }
 
+func templateFuncs() template.FuncMap {
+	return map[string]interface{}{
+		// dockerDomain returns the domain/host information for the provided
+		// docker image
+		"dockerDomain": func(in string) string {
```

Comment:
Does it have to be `dockerFoo`? Could this be `imageRefDomain` or something a bit more OCI-friendly? 😉

Reply:
Was chatting on slack, but I was suggesting since we had interpolation support in heredocs, I prefer if we use regular string functions rather than template functions.

Along with the [array support proposal](https://github.com/openllb/hlb/issues/211), you could do something like:
```hlb
[]string dockerRegistries() {
	"titusregistry.us-east-1.netflix.net"
	"titusregistry.us-west-2.netflix.net"
}

[]string pushTargets(string ref) {
	for (string registry in dockerRegistries) {
		"${registry}/${dockerPath(ref)}:${dockerTag(ref)}
	}
}

fs titusPush(string ref) {
	for (string target in pushTargets(ref)) {
		dockerPush target
	}
}
```